src/test/java/com/wikia/webdriver/testcases/adstests/TestAdsVelesTracking.java

```diff
@@ -0,0 +1,148 @@
+  )
+  public void adsTrackingVelesErrorTracked(
+    final Page page,
+    final Map<String, String> positionsAndPrices,
+    final Map<String, DefaultHttpResponse> mockRules
+  ) {
+    mockResponses(mockRules);
+    adsTrackingVelesTracked(page, positionsAndPrices);
+  }
+
+  private void mockResponses(Map<String, DefaultHttpResponse> mockRules) {
+    if (mockRules != null && !mockRules.isEmpty()) {
+      networkTrafficInterceptor.addRequestFilter((request, contents, messageInfo) -> {
+
+        for (Map.Entry<String, DefaultHttpResponse> rule : mockRules.entrySet()) {
+          if (request.getUri().matches(rule.getKey())) {
+            return rule.getValue();
+          }
+        }
+
+        return null;
+      });
+    }
+  }
+
+  private Set<HarEntry> findEntriesByUrlPart(AdsBaseObject pageObject, final String s, final Map<String, String> positionsAndPrices) {
+    for(String slotName: positionsAndPrices.keySet()) {
+      waitForSlotTrackingRequest(pageObject, slotName);
+    }
+
+    return networkTrafficInterceptor.getHar().getLog().getEntries().stream()
+        .filter(entry -> entry.getRequest().getUrl().contains(s))
+        .collect(toSet());
+  }
+
+  private void waitForSlotTrackingRequest(AdsBaseObject pageObject, String slotName) {
+    final String pattern = String.format(".*%s.*%s=%s.*", VELES_TRACKING_REQUEST_PART, VELES_TRACKING_POSITION_PARAMETER, slotName);
+    pageObject.wait.forSuccessfulResponseByUrlPattern(networkTrafficInterceptor, pattern);
+  }
+
+  private boolean wasVelesTrackedIn(final Set<HarEntry> entries, final Map<String, String> positionsAndPrices) {
+    boolean result = false;
+
+    if (!entries.isEmpty()) {
+
+      result = entries.stream()
+          .map(entry -> URLEncodedUtils.parse(entry.getRequest().getUrl(), StandardCharsets.UTF_8))
+          .map(VelesTrackingInfo::fromParameters)
+          .allMatch(trackingInfo -> validatePricesIn(trackingInfo, positionsAndPrices));
+    }
+
+    return result;
+  }
+
+  private boolean validatePricesIn(final VelesTrackingInfo trackingInfo, final Map<String, String> positionsAndPrices) {
+    return positionsAndPrices.containsKey(trackingInfo.getPosition())
+        ? positionsAndPrices.get(trackingInfo.getPosition()).equals(trackingInfo.getPrice())
+        : "".equals(trackingInfo.getPrice());
+  }
+
+  private static class VelesTrackingInfo {
+    private final String position;
+
+    private final String price;
+
+    private VelesTrackingInfo(String position, String price) {
+      this.position = position;
+      this.price = price;
+    }
+
+    String getPosition() {
+      return position;
+    }
+
+    String getPrice() {
+      return price;
+    }
+
+    @Override
+    public String toString() {
+      return "VelesTrackingInfo{" +
+          "position='" + position + '\'' +
+          ", price='" + price + '\'' +
+          '}';
+    }
+
+    private static VelesTrackingInfo fromParameters(final List<NameValuePair> parameters) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this unused private "fromParameters" method. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar-s4/coding_rules#rule_key=squid%3AUnusedPrivateMethod)